test-data/unit/check-final.test

```diff
@@ -741,9 +773,12 @@ class B:
         self.x = 2
 class C(A, B): ...  # E: Cannot override writable attribute "x" with a final one
 class D(B, A): ...  # E: Cannot override final attribute "x" (previously declared in base class "A")
-C.x = 3  # E: Cannot assign to final attribute "x"
+C.x = 3  # E: Cannot assign to final attribute "x" \
+         # E: Incompatible types in assignment (expression has type "Literal[3]", variable has type "Literal[1]")
 D.x = 3  # E: Cannot assign to final attribute "x"
-C().x = 4  # E: Cannot assign to final attribute "x"
+C().x = 4  # E: Cannot assign to final attribute "x" \
+           # E: Incompatible types in assignment (expression has type "Literal[4]", variable has type "Literal[1]")
+# TODO: this missing error is a sign of a bug!!
```

Comment:
I also wrote up https://github.com/python/mypy/issues/15640 while doing this (obviously).

I still need to make a minimized issue for this `TODO`.